### PR TITLE
Adds a Terrform Module for External Storage

### PR DESCRIPTION
**Pull request by jasonrogena**
Adds a terraform module for provisioning external storage for VMs on AWS.

### CODE REVIEW

File: roles/opensrp/tasks/postgresql.yml [hunk 1]
```diff
@@ -1,16 +1,24 @@
 ---
+- name: Ensure checkout directory is empty
+  file:
+    state: absent
+    path: "{{ opensrp_local_checkout_path }}"
+  tags:
+    - postgresql
+
+- name: Checking out codebase
```
Comment (morrismukiri) [inline roles/opensrp/tasks/postgresql.yml:9]: Should this be in this file?

File: terraform/modules/providers/aws/virtual-machine-storage/storage.tf [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  tags = {
+    Name            = "${var.vms_name}-${count.index}"
+    OwnerList       = var.vms_owner
+    ProjectList     = var.vms_project
+    EndDate         = var.vms_end_date
+    EnvironmentList = var.vms_env
+  }
+}
+
+resource "aws_volume_attachment" "volume_attachment" {
```
Comment (morrismukiri) [inline terraform/modules/providers/aws/virtual-machine-storage/storage.tf:17]: Why Do we need an attached volume and an ebs volume on a host?
From Terraform docs

> NOTE on EBS block devices: If you use ebs_block_device on an aws_instance, Terraform will assume management over the full set of non-root EBS block devices for the instance, and treats additional block devices as drift. For this reason, ebs_block_device cannot be mixed with external aws_ebs_volume + aws_ebs_volume_attachment resources for a given instance.

Comment (jasonrogena) [reply terraform/modules/providers/aws/virtual-machine-storage/storage.tf:17]: Due to how loops work on HCL, that's the only way I could find to allow the VM module spin up instances that don't need external volumes instances that need external storage.

@morrismukiri suggestions on doing it in a better way are welcome.

Comment (jasonrogena) [reply terraform/modules/providers/aws/virtual-machine-storage/storage.tf]: FYI, I haven't used EBS block devices.

File: roles/opensrp/tasks/postgresql.yml (hunk 1 above)
Comment (jasonrogena) [reply roles/opensrp/tasks/postgresql.yml]: Yes. You need to clone the repository in the OpenSRP Server host before running the migrations from the host.

Comment (jasonrogena) [reply roles/opensrp/tasks/postgresql.yml]: We previously were cloning the repo in the host running Ansible then copying over the directories used for migrations. That proved to be slow (and would timeout sometimes), hence why I suggest we clone the repo directly on the app host before running the migration. This isn't the clone that the OpenSRP server WAR file is built using.

Comment (jasonrogena) [reply roles/opensrp/tasks/postgresql.yml:9]: We previously were cloning the repo in the host running Ansible then copying over the directories used for migrations, before running the migrations. That proved to be slow (and would timeout sometimes), hence why I suggest we clone the repo directly on the app host for the purpose of running the migrations.

Comment (jasonrogena) [reply roles/opensrp/tasks/postgresql.yml:9]: The repo is now cloned twice:
1. On the host running Ansible, right before building the WAR file
2. On the app host, right before migrations are run from there

File: terraform/modules/providers/aws/virtual-machine-storage/storage.tf (hunk 2 above)
Comment (morrismukiri) [reply terraform/modules/providers/aws/virtual-machine-storage/storage.tf:17]: Alright. I got you.

File: roles/opensrp/tasks/postgresql.yml (hunk 1 above)
Comment (morrismukiri) [reply roles/opensrp/tasks/postgresql.yml:9]: Alright. Makes sense then. Would we want to cleanup after running the migrations?

Comment (jasonrogena) [reply roles/opensrp/tasks/postgresql.yml:9]: Added the task.

**Review by morrismukiri — APPROVED** [review]

**Pull request merged**